shared/src/commonMain/kotlin/db/database/user/UserDatabase.kt

```diff
@@ -0,0 +1,26 @@
+package db.database.user
+
+import app.cash.sqldelight.db.SqlDriver
+import db.chargehub.ChargeHubDb
+import db.chargehub.User
+import db.networking.request.CreateUserRequest
+
+class UserDatabase(sqlDriver: SqlDriver) {
+    private val database = ChargeHubDb(sqlDriver)
+    private val dbQuery = database.userDbQueries
+
+    fun getAllUsers(): List<User> {
+        return dbQuery.getAllUsers().executeAsList()
+    }
+
+    fun createUser(user: CreateUserRequest) {
+        dbQuery.insertUser(
```

Comment:
Is it possible to insert the whole user or is there a reason you insert the fields seperately?

Reply:
I think it needs the fields because this is the "auto generated" method built from the sqlQuery, so it needs to know what to insert where. I checked through some example repositories, and the ones I checked all seemed to do it like this, so not sure if its possible to insert the entire object somehow.